packages/firestore/src/remote/remote_store.ts

```diff
@@ -691,46 +678,17 @@ export class RemoteStore implements TargetMetadataProvider {
       );
     }
 
-    // If the write stream closed due to an error, invoke the error callbacks if
-    // there are pending writes.
-    if (error && this.writePipeline.length > 0) {
-      if (this.writeStream.handshakeComplete) {
-        // This error affects the actual write.
-        await this.handleWriteError(error!);
-      } else {
-        // If there was an error before the handshake has finished, it's
-        // possible that the server is unable to process the stream token
-        // we're sending. (Perhaps it's too old?)
-        await this.handleHandshakeError(error!);
-      }
-
-      // The write stream might have been started by refilling the write
-      // pipeline for failed writes
-      if (this.shouldStartWriteStream()) {
-        this.startWriteStream();
-      }
+    // If the write stream closed after the write handshake completes, a write
+    // operation failed and we fail the pending operation.
+    if (error && this.writeStream.handshakeComplete) {
+      // This error affects the actual write.
+      await this.handleWriteError(error!);
```

Comment:
We're still tracking the `lastStreamToken` and it's possible that if we're restarting the write stream after some successful writes, the server could reject the stream token we supply. This means we still need to handle the handshake error.

Reply:
I'll add the handshake code back if you prefer (it might be safer to reduce the churn here), but what do you think about the fix in b530cd58d?

Reply:
I have no strong preference on the general strategy, though it seems like clearing when the stream starts could be more bullet proof.

Reply:
I modified it to match the assignment of `handshakeComplete_`, which is now true only if the stream token is non empty.